evalml/tests/automl_tests/test_automl.py

```diff
@@ -1715,3 +1715,65 @@ def test_automl_error_callback(mock_fit, mock_score, X_y_binary, caplog):
     assert len(automl._results['errors']) == 1  # Raises exception at first error
     for e in automl._results['errors']:
         assert str(e) == msg
+
+
+@patch('evalml.pipelines.BinaryClassificationPipeline.score')
+@patch('evalml.pipelines.BinaryClassificationPipeline.fit')
+def test_automl_woodwork_user_types_preserved(mock_fit, mock_score, X_y_binary):
+    X, y = X_y_binary
+    X = pd.DataFrame(X)
+    new_col = np.zeros(len(X))
+    new_col[:int(len(new_col) / 2)] = 1
+    X['cat col'] = pd.Series(new_col)
+    X = ww.DataTable(X, semantic_tags={'cat col': 'category'}, logical_types={'cat col': ww.logical_types.Categorical})
```

Comment:
Is it necessary to pass in the semantic/logical info here? Woodwork should infer that "cat col" is binary/categorical, yes?

Reply:
@dsherry Without passing it in directly, I think it infers the 'category' semantic tag but also numeric/categorical logic tags. Thus, I purposely set this to be what woodwork would *not* infer, and check that the user-specified edits are preserved :)

Reply:
@angela97lin One Woodwork side note here - you can pass in the LogicalType specifiers as camel-case or snake-case strings as well, so you don't need to use the full `ww.logical_types.Categorical` syntax, but you could just pass `"Categorical"` instead - might keep the code a tad bit cleaner.

Reply:
@thehomebrewnerd Oo awesome, thank you for this! Indeed, would make things cleaner--will update! 😁

Reply:
The string syntax also helps in situations where you would otherwise have to `import ww` or similar in a file just for the sake of setting the types, and lets you eliminate that import.